src/PhenX.EntityFrameworkCore.BulkInsert/Extensions/DbContextExtensions.cs

```diff
@@ -69,4 +71,17 @@ internal static async Task<ConnectionInfo> GetConnection(
 
         return new ConnectionInfo(connection, wasClosed, transaction, wasBegan);
     }
+
+    /// <summary>
+    /// Tells if the current provider is the specified provider type.
+    /// </summary>
+    internal static bool IsProvider(this DbContext context, ProviderType providerType)
+    {
+        if (context.Database.ProviderName == null)
+        {
+            throw new InvalidOperationException("Database provider name is null.");
+        }
+
+        return context.Database.ProviderName.Contains(providerType.ToString(), StringComparison.OrdinalIgnoreCase);
```

Comment:
If you already define the enum, you could also add this to the Provider as property and check just the registered provider.